keps/sig-node/3000-sizing-providers/README.md

```diff
@@ -0,0 +1,322 @@
+  - [ ] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: KubeletSizingProvider
+    - Components depending on the feature gate: kubelet
+
+* **Does enabling the feature change any default behavior?**
+  No, use of this feature still requires extra flags enabled on the kubelet.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Yes, as long as kubelet does not specify the flags `--sizing-provider-config` and `--sizing-provider-bin-dir`.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  Kubelet will continue to invoke exec plugins. No state is stored for this feature to function.
+
+* **Are there any tests for feature enablement/disablement?**
+  Yes.
+
+### Rollout, Upgrade and Rollback Planning
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can a rollout fail? Can it impact already running workloads?**
+
+TBD for beta.
+
+* **What specific metrics should inform a rollback?**
+
+TBD for beta.
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+
+TBD for beta.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs,
+fields of API types, flags, etc.?**
+
+TBD for beta.
+
+
+
+### Scalability
+
+_For alpha, this section is encouraged: reviewers should consider these questions
+and attempt to answer them._
+
+_For beta, this section is required: reviewers must answer these questions._
+
+_For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field._
+
+* **Will enabling / using this feature result in any new API calls?**
+
+No
+
+* **Will enabling / using this feature result in introducing new API types?**
+
+It will add a new kubelet-level API. This API only contains a TypeMeta though and is
+not an object.
+
+* **Will enabling / using this feature result in any new calls to the cloud
+provider?**
+
+No, but a plugin implementation may choose to make API calls to a cloud provider.
+
+* **Will enabling / using this feature result in increasing size or count of
+the existing API objects?**
+
+No.
+
+* **Will enabling / using this feature result in increasing time taken by any
+operations covered by [existing SLIs/SLOs]?**
+
+Use of the exec plugin may increase the startup time if the exec plugin invoked
+by kubelet takes a long time.
+
+* **Will enabling / using this feature result in non-negligible increase of
+resource usage (CPU, RAM, disk, IO, ...) in any components?**
+
+Possibly, it depends on how long it takes for the exec plugin to return during kubelet startup.
+
+## Implementation History
+
+
+## Drawbacks
+
+* exec plugins may be expensive to invoke by kubelet during the startup.
+* a poorly implemented exec plugin may halt the startup for the kubelet.
+
+## Alternatives
+
+1. add a built-in sizing provider in-tree.
+
```

Comment:
Why not add an HTTP endpoint with a standardized schema that one can POST the desired values to, with the appropriate authn and authz?

Reply:
Please make sure you add a consideration and discussion of this to the Alternatives section. Right now it has one bullet about an in-tree implementation, but doesn't consider pros and cons. Pros and cons of all alternative solutions should be outlined.

Reply:
Thanks @ehashman